frappe/desk/query_report.py

```diff
@@ -722,6 +726,36 @@ def has_match(
 	return resultant_match
 
 
+@site_cache
```

Comment:
site_cache isn't invalidated, so it won't get invalidated when permissions change. 

@request_cache is what you want here most likely. 
Both @redis_cache and @site_cache will require you to invalidate them when permissions change.

Reply:
Nope. That's just single worker, won't work in prod.